cmd/admin-heal-ops.go

```diff
@@ -766,28 +766,23 @@ func (h *healSequence) healFromSourceCh() {
 	h.healItemsFromSourceCh()
 }
 
-func (h *healSequence) healDiskMeta() error {
-	// Start with format healing
-	if err := h.healDiskFormat(); err != nil {
-		return err
-	}
-
+func (h *healSequence) healDiskMeta(objAPI ObjectLayer) error {
 	// Start healing the config prefix.
-	if err := h.healMinioSysMeta(minioConfigPrefix)(); err != nil {
+	if err := h.healMinioSysMeta(objAPI, minioConfigPrefix)(); err != nil {
 		return err
 	}
 
 	// Start healing the bucket config prefix.
-	return h.healMinioSysMeta(bucketConfigPrefix)()
+	return h.healMinioSysMeta(objAPI, bucketConfigPrefix)()
```

Comment:
Whoa. Are we already doing this?

Not all data in there can be healed. Both crawler and the list cache has per-set data, meaning *the same files will be present in each set with unique data*.

That is a huge issue if these are "healing" across sets.

Reply:
@klauspost not sure what you are talking about?

Reply:
@harshavardhana Data is unique per set, meaning the data in `.minio.sys/buckets/.usage-cache.bin` is not the same for set 0 as for set 1, etc. This is because crawling is separate per set and only aggregated to the total for the cluster.

So unless this healing is strictly per set it could have some bad side-effects.